topo/node/cptx/cptx.go

```diff
@@ -4,220 +4,228 @@
 package cptx
 
 import (
-	"context"
-	"errors"
-	"fmt"
-	"io"
-	"io/ioutil"
-	"time"
-
-	topopb "github.com/google/kne/proto/topo"
-	"github.com/google/kne/topo/node"
-	scraplibase "github.com/scrapli/scrapligo/driver/base"
-	scraplicore "github.com/scrapli/scrapligo/driver/core"
-	scraplinetwork "github.com/scrapli/scrapligo/driver/network"
-	scraplitransport "github.com/scrapli/scrapligo/transport"
-	scraplicfg "github.com/scrapli/scrapligo/cfg"
-	log "github.com/sirupsen/logrus"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-	"google.golang.org/protobuf/proto"
+    "context"
+    "errors"
+    "fmt"
+    "io"
+    "io/ioutil"
+    "time"
+
+    topopb "github.com/google/kne/proto/topo"
+    "github.com/google/kne/topo/node"
+    scraplibase "github.com/scrapli/scrapligo/driver/base"
+    scraplicore "github.com/scrapli/scrapligo/driver/core"
+    scraplinetwork "github.com/scrapli/scrapligo/driver/network"
+    scraplitransport "github.com/scrapli/scrapligo/transport"
+    scraplicfg "github.com/scrapli/scrapligo/cfg"
+    log "github.com/sirupsen/logrus"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/status"
+    "google.golang.org/protobuf/proto"
 )
 
 // ErrIncompatibleCliConn raised when an invalid scrapligo cli transport type is found.
 var ErrIncompatibleCliConn = errors.New("incompatible cli connection in use")
-const max_retries = 10
+// Approx timeout while we wait for cli to get ready
+const timeout = 500
 
 func New(pb *topopb.Node) (node.Implementation, error) {
-	cfg := defaults(pb)
-	proto.Merge(cfg, pb)
-	node.FixServices(cfg)
-	return &Node{
-		pb: cfg,
-	}, nil
+    cfg := defaults(pb)
+    proto.Merge(cfg, pb)
+    node.FixServices(cfg)
+    return &Node{
+        pb: cfg,
+    }, nil
 }
 
 type Node struct {
-	pb *topopb.Node
-	cliConn *scraplinetwork.Driver
+    pb *topopb.Node
+    cliConn *scraplinetwork.Driver
 }
 
 func (n *Node) Proto() *topopb.Node {
-	return n.pb
+    return n.pb
 }
 
 // WaitCLIReady attempts to open the transport channel towards a Network OS and perform scrapligo OnOpen actions
 // for a given platform. Retries with exponential backoff.
-func (n *Node) WaitCLIReady(sleep time.Duration, attempts int) error {
+func (n *Node) WaitCLIReady(ctx context.Context) error {
     var err error = nil
```

Comment:
var err error

default is nil no need to set it

Reply:
Oops, I'll take care of it.

Reply:
Done.